executor/src/witgen/jit/witgen_inference.rs

```diff
@@ -203,24 +203,71 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> WitgenInference<'a, T, F
             row_offset,
             rhs: VariableOrValue::Variable(variable),
         });
-        self.process_assignments();
+        self.process_assignments().unwrap();
     }
 
+    /// Processes an equality constraint.
+    /// If this returns an error, it means we have conflicting constraints.
     fn process_equality_on_row(
         &self,
         lhs: &Expression<T>,
         offset: i32,
-        rhs: AffineSymbolicExpression<T, Variable>,
-    ) -> ProcessResult<T, Variable> {
-        if let Some(r) = self.evaluate(lhs, offset) {
-            // TODO propagate or report error properly.
-            // If solve returns an error, it means that the constraint is conflicting.
-            // In the future, we might run this in a runtime-conditional, so an error
-            // could just mean that this case cannot happen in practice.
-            (r - rhs).solve().unwrap()
-        } else {
-            ProcessResult::empty()
+        rhs: &VariableOrValue<T, Variable>,
+    ) -> Result<ProcessResult<T, Variable>, EvalError<T>> {
+        // First we try to find a new assignment to a variable in the equality.
+
+        let evaluator = Evaluator::new(self);
+        let Some(lhs_evaluated) = evaluator.evaluate(lhs, offset) else {
+            return Ok(ProcessResult::empty());
+        };
+
+        let rhs_evaluated = match rhs {
+            VariableOrValue::Variable(v) => evaluator.evaluate_variable(v.clone()),
+            VariableOrValue::Value(v) => (*v).into(),
+        };
+
+        let result = (lhs_evaluated - rhs_evaluated)
+            .solve()
+            .map_err(|e| match e {
+                EvalError::ConflictingRangeConstraints | EvalError::ConstraintUnsatisfiable(_) => e,
```

Comment:
Maybe the error type should be something different then? The `EvalError` type is kind of complicated and contains a lot of errors that are not applicable in the JIT context, and now it's exposed to users of `WitgenInference`. Right now, the error type could just be `()`, and we could improve it later to include information which constraint failed and what the referenced values were.